juniper_codegen/src/impl_object.rs

```diff
@@ -52,19 +52,23 @@ pub fn build_object(args: TokenStream, body: TokenStream, is_internal: bool) ->
                 .unwrap()
                 .ident
                 .to_string(),
-            syn::Type::Reference(ref reference) => match &*reference.elem {
-                syn::Type::Path(ref type_path) => type_path
-                    .path
-                    .segments
-                    .iter()
-                    .last()
-                    .unwrap()
-                    .ident
-                    .to_string(),
-                _ => {
-                    panic!("Could not determine a name for the object type: specify one with #[juniper::object(name = \"SomeName\")");
-                }
-            },
+            syn::Type::Reference(ref reference) => {
+                let path = match &*reference.elem {
+                    syn::Type::Path(ref type_path) => &type_path.path,
```

Comment:
Do we not need to get the last segment anymore?

Reply:
This just extracts the path from the reference. Getting the last segment is done on [line 70](https://github.com/graphql-rust/juniper/pull/385/files/300d2fd352aaf0bb099e6de427fb7c857056eaf1#diff-0c158a3c35852b789d818d59989ccb55R70).